src/components/slider/draw.js

```diff
@@ -0,0 +1,507 @@
+    sliderOpts.labelStride = Math.max(1, Math.ceil(computedSpacePerLabel / availableSpacePerLabel));
+    sliderOpts.labelHeight = labelHeight;
+
+    // Hard-code this for now:
+    sliderOpts.height = constants.tickOffset + constants.tickLength + sliderOpts.labelHeight + sliderOpts.ypad * 2;
+
+    var xanchor = 'left';
+    if(anchorUtils.isRightAnchor(sliderOpts)) {
+        sliderOpts.lx -= sliderOpts.outerLength;
+        xanchor = 'right';
+    }
+    if(anchorUtils.isCenterAnchor(sliderOpts)) {
+        sliderOpts.lx -= sliderOpts.outerLength / 2;
+        xanchor = 'center';
+    }
+
+    var yanchor = 'top';
+    if(anchorUtils.isBottomAnchor(sliderOpts)) {
+        sliderOpts.ly -= sliderOpts.height;
+        yanchor = 'bottom';
+    }
+    if(anchorUtils.isMiddleAnchor(sliderOpts)) {
+        sliderOpts.ly -= sliderOpts.height / 2;
+        yanchor = 'middle';
+    }
+
+    sliderOpts.outerLength = Math.ceil(sliderOpts.outerLength);
+    sliderOpts.height = Math.ceil(sliderOpts.height);
+    sliderOpts.lx = Math.round(sliderOpts.lx);
+    sliderOpts.ly = Math.round(sliderOpts.ly);
+
+    Plots.autoMargin(gd, constants.autoMarginIdRoot + sliderOpts._index, {
+        x: sliderOpts.x,
+        y: sliderOpts.y,
+        l: sliderOpts.outerLength * ({right: 1, center: 0.5}[xanchor] || 0),
+        r: sliderOpts.outerLength * ({left: 1, center: 0.5}[xanchor] || 0),
+        b: sliderOpts.height * ({top: 1, middle: 0.5}[yanchor] || 0),
+        t: sliderOpts.height * ({bottom: 1, middle: 0.5}[yanchor] || 0)
+    });
+}
+
+function drawSlider(gd, sliderGroup, sliderOpts) {
+    // These are carefully ordered for proper z-ordering:
+    sliderGroup
+        .call(drawRail, sliderOpts)
+        .call(drawLabelGroup, sliderOpts)
+        .call(drawTicks, sliderOpts)
+        .call(drawTouchRect, gd, sliderOpts)
+        .call(drawGrip, gd, sliderOpts);
+
+    // Position the rectangle:
+    Lib.setTranslate(sliderGroup, sliderOpts.lx + sliderOpts.xpad, sliderOpts.ly + sliderOpts.ypad);
+
+    removeListeners(gd, sliderGroup, sliderOpts);
+    attachListeners(gd, sliderGroup, sliderOpts);
+
+    setActive(gd, sliderGroup, sliderOpts, sliderOpts.active, true, false);
+}
+
+function removeListeners(gd, sliderGroup, sliderOpts) {
+    var listeners = sliderOpts._input.listeners;
+    var eventNames = sliderOpts._input.eventNames;
+    if(!Array.isArray(listeners) || !Array.isArray(eventNames)) return;
+    while(listeners.length) {
+        gd._removeInternalListener(eventNames.pop(), listeners.pop());
+    }
+}
+
+function attachListeners(gd, sliderGroup, sliderOpts) {
+    var listeners = sliderOpts._input.listeners = [];
+    var eventNames = sliderOpts._input.eventNames = [];
+
+    function makeListener(eventname, updatevalue) {
+        return function(data) {
+            var value = data;
+            if(updatevalue) {
+                value = Lib.nestedProperty(data, updatevalue).get();
```

Comment:
Ah, that's a good question… It may well not. I simply assumed (hoped?) it would. Thanks for noticing.